docs: add changelog and update readme (#775)
vercel/styled-jsx: Changelog.md

# Changelog

## [5.0.0]

### Features

- Introduce contextual styles (#744)
- Opt-in react 18 insertion effect hook when available (#753)
- Fallback to module level registry in browser (#768)

### Improvements

- Make JSXStyle return a noop if the registry context is not provided (#749)
- Fix typings of `nonce` property
- Pre-compile dependencies to reduce install size/time (#770)

### BREAKING CHANGES

#### APIs

- `styled-jsx/server` import path is deprecated
- `flush` and `flushToHTML` from `styled-jsx/server` APIs are deprecated
- New component `<StyledRegistry>` is introduced
- New APIs `useStyleRegistry` and `createStyleRegistry` are introduced

#### Usage

> If you're only use styled-jsx in pure client side, nothing will effect you.

Reviewer: ```suggestion
If you're only using styled-jsx purely client side, nothing will effect you.
```

> If you're using styled-jsx inside Next.js without customization, Next.js will manage the new gaps and it's same as previous.

Reviewer: ```suggestion
If you're using styled-jsx inside Next.js without customization, Next.js will automatically handle the changes for you.
```
